### PR TITLE
Support KafkaConnect Records via Converters

### PR DESCRIPTION
**Pull request by renatomefi**
This allows the Task to be able to create SourceRecords which might be coming from an already generated Kafka Connect Record.

More context: https://github.com/salesforce/mirus/issues/25#issuecomment-444964030

### CODE REVIEW

File: src/main/java/com/salesforce/mirus/MirusSourceTask.java [hunk 1]
```diff
@@ -201,12 +229,12 @@ private SourceRecord toSourceRecord(ConsumerRecord<byte[], byte[]> consumerRecor
...
         sourcePartition,
         offsetMap(consumerRecord.offset() + 1),
-        destinationTopicNamePrefix + consumerRecord.topic() + destinationTopicNameSuffix,
+        topic,
         enablePartitionMatching ? consumerRecord.partition() : null,
-        Schema.OPTIONAL_BYTES_SCHEMA,
-        consumerRecord.key(),
-        Schema.OPTIONAL_BYTES_SCHEMA,
-        consumerRecord.value(),
+        keySchema,
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/MirusSourceTask.java]: You can simplify this change by calling `keyAndSchema.schema()`,  `keyAndSchema.value()`, etc. directly here.  That will allow you to remove the local variable definitions above.

File: src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java [hunk 2]
```diff
@@ -62,6 +62,24 @@
       false,
       ConfigDef.Importance.MEDIUM,
       "Ensures records are written to the destination partition with the same identifier as the source partition"),
+  SOURCE_KEY_CONVERTER(
+      "source.key.converter",
+      ConfigDef.Type.CLASS,
+      "org.apache.kafka.connect.converters.ByteArrayConverter",
+      ConfigDef.Importance.MEDIUM,
+      "Allows to convert key data from the source just like as a SinkConnector would do"),
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java]: Please change to "Converter class to apply to source record keys".

File: src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java [hunk 3]
```diff
@@ -62,6 +62,24 @@
...
+  SOURCE_VALUE_CONVERTER(
+      "source.value.converter",
+      ConfigDef.Type.CLASS,
+      "org.apache.kafka.connect.converters.ByteArrayConverter",
+      ConfigDef.Importance.MEDIUM,
+      "Allows to convert key data from the source just like as a SinkConnector would do"),
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java]: Please change to "Converter class to apply to source record values".

File: src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java [hunk 4]
```diff
@@ -62,6 +62,24 @@
...
+  SOURCE_HEADER_CONVERTER(
+      "source.header.converter",
+      ConfigDef.Type.CLASS,
+      "org.apache.kafka.connect.converters.ByteArrayConverter",
+      ConfigDef.Importance.MEDIUM,
+      "Allows to convert header data from the source just like as a SinkConnector would do"),
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java]: Please change to "Converter class to apply to source record headers".

**Comment by renatomefi** [comment]
@pdavidson100 thanks for the review! I've applied all your suggestions! :+1:

**Pull request merged**